### PR TITLE
Feature/isdb 283 group framework

### PR DESCRIPTION
**Pull request by Murshadhaque**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanA58Service.java [hunk 1]
```diff
@@ -373,4 +387,63 @@ private void buildWelshSpokenPreferences(Map<String, String> inputFieldsMap, Map
             populatedMap.put(CHILD_WELSH_LANGUAGE_PREFERENCE_CCD, inputFieldsMap.get(CHILD_WELSH_LANGUAGE_PREFERENCE));
         }
     }
+
+    private void updateGroupErrorsAndWarnings(BulkScanValidationRequest bulkRequest, FormType formType,
+                                              BulkScanValidationResponse bulkScanValidationResponse) {
```
Comment (hmcts-kodatala) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanA58Service.java:392]: add java doc which give part of form page and section will cover the validation

File: src/test/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanA58GroupFieldsTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    private static final String A58_GROUP_FIELD_REQUEST_PATH =
+        "classpath:request/bulk-scan-a58-group-input.json";
+    private static final String A58_GROUP_FIELD_REQUEST_WARNING_PATH =
+        "classpath:request/bulk-scan-a58-group-warning-input.json";
+    private static final String A58_GROUP_FIELD_REQUEST_ERROR_PATH =
+        "classpath:request/bulk-scan-a58-group-error-input.json";
+
+    @Spy
+    @Autowired
+    BulkScanA58Service bulkScanValidationService;
```
Comment (hmcts-kodatala) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanA58GroupFieldsTest.java:35]: @Spy not required and autowired is there

File: src/test/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanA58GroupFieldsTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    @Test
+    @DisplayName("A58 Group Fields form validation success scenario")
+    void testA58PostPlacementApplicationValidationErrors() throws IOException {
+        BulkScanValidationRequest bulkScanValidationRequest = mapper
+            .readValue(readFileFrom(A58_GROUP_FIELD_REQUEST_ERROR_PATH), BulkScanValidationRequest.class);
+        BulkScanValidationResponse res = bulkScanValidationService.validate(bulkScanValidationRequest);
+        assertEquals(Status.ERRORS, res.status);
+    }
```
Comment (hmcts-kodatala) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanA58GroupFieldsTest.java:62]: need to update the test case descriptions to error/warnings???

**Review by hmcts-kodatala — COMMENTED** [review]
added review comments

**Review by hmcts-kodatala — APPROVED** [review]
OK

**Review by rahulsanjava19 — APPROVED** [review]

**Pull request merged**